controllers/clustercache/cluster_accessor_client.go

```diff
@@ -263,7 +265,7 @@ func createCachedClient(ctx context.Context, clusterAccessorConfig *clusterAcces
 	defer cacheSyncCtxCancel()
 	if !cache.WaitForCacheSync(cacheSyncCtx) {
 		cache.Stop()
-		return nil, nil, errors.Wrapf(cacheCtx.Err(), "error when waiting for cache to sync")
+		return nil, nil, fmt.Errorf("error when waiting for cache to sync: %w", cacheSyncCtx.Err())
```

Comment:
Before we used errors.Wrap *and* with the wrong context. The result was that if the cache actually never synced. cacheCtx.Err() was nil and then errors.Wrapf also returned nil. This leads to panics down the line